jetty-util/src/main/java/org/eclipse/jetty/util/Scanner.java

```diff
@@ -544,30 +508,22 @@ public void removeListener(Listener listener)
     @Override
     public void doStart()
     {
-        try (AutoLock l = _lock.lock())
-        {
-            if (_running)
-                return;
-
-            _running = true;
-            if (LOG.isDebugEnabled())
-                LOG.debug("Scanner start: rprtExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
-                    _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
+        if (LOG.isDebugEnabled())
+            LOG.debug("Scanner start: rprtExists={}, depth={}, rprtDirs={}, interval={}, filter={}, scannables={}",
+                _reportExisting, _scanDepth, _reportDirs, _scanInterval, _filter, _scannables);
 
-            if (_reportExisting)
-            {
-                // if files exist at startup, report them
-                scan();
-                scan(); // scan twice so files reported as stable
-            }
-            else
-            {
-                //just register the list of existing files and only report changes
-                scanFiles();
-                _prevScan.putAll(_currentScan);
-            }
-            schedule();
+        if (_reportExisting)
+        {
+            // if files exist at startup, report them
+            scan();
+            scan(); // scan twice so files reported as stable
+        }
+        else
+        {
+            //just register the list of existing files and only report changes
+            _prevScan = scanFiles();
         }
+        schedule();
     }
 
     public TimerTask newTimerTask()
```

Comment:
Make it `private` and replace with Jetty's `Scheduler`.

Reply:
Done.